### PR TITLE
plug minor leaks

### PR DESCRIPTION
**Pull request by the8472**
This should fix the lingering sandbox when the addon gets updated

### CODE REVIEW

File: resource/modules/content/utils/api.jsm [hunk 1]
```diff
@@ -1,6 +1,7 @@
-// VERSION 1.1.0
+// VERSION 1.1.1
 
 this.api = {
+	QueryInterface: XPCOMUtils.generateQI([Ci.nsISupports, Ci.nsISupportsWeakReference, Ci.nsIDOMEventListener]),
```
Comment (Quicksaver) [inline resource/modules/content/utils/api.jsm]: Please just quickly add a new line after this.

File: resource/modules/content/utils/api.jsm [hunk 2]
```diff
@@ -41,6 +42,10 @@ this.api = {
 
 	onFrameDeleted: function(frame) {
 		frame.removeEventListener('DOMContentLoaded', this);
+		if(frame.content) {
+			frame.content.removeEventListener('load', this);
```
Comment (Quicksaver) [inline resource/modules/content/utils/api.jsm]: Nice catch!

**Comment by Quicksaver** [comment]
Otherwise, very nice catches! Thank you for addressing them. :)

**Comment by Quicksaver** [comment]
I'm curious though, why is the QueryInterface reference necessary in this case?

**Comment by the8472** [comment]
Maybe not necessary, I just tried both approaches at once.

**Comment by Quicksaver** [comment]
Can you please add a comment above it saying something about it ("this may or may not be necessary because of reasons" or something like that), otherwise it'll be confusing later trying to figure out why it's there.

**Pull request merged**